AzureMonitor/Scripts/Create-AzDiagPolicy.ps1

```diff
@@ -1414,55 +1798,155 @@ IF($($ExportEH) -or ($ExportLA))
             if($ExportEH)
             {
-                $RPVar = Parse-ResourceType -resourceType $Type.ResourceType -sinkDest "EH"
-                $PolicyJSON = Update-EventHubJSON -resourceType $Type.ResourceType -metricsArray $metricsArray -logsArray $logsArray -nameField $RPVar[1]
-                write-host "Exporting Event Hub Custom Azure Policy for resourceType: $($Type.ResourceType)" -ForegroundColor Yellow
+                $sinkDest = "EH"
+                $RPVar = Parse-ResourceType -resourceType $Type.ResourceType -sinkDest $sinkDest
+                # If we have a kind for the resourceType let's add that to the policy evaluation rules (and add it to the displayname)
+                if($Type.Kind)
+                {
+                    $PolicyResourceDisplayName = "Apply Diagnostic Settings for $($Type.ResourceType) `($($Type.Kind)`) to a Regional Event Hub"
+                }
+                elseif (!($Type.Kind))
+                {
+                    $PolicyResourceDisplayName = "Apply Diagnostic Settings for $($Type.ResourceType) to a Regional Event Hub"
+                }
+                
+                # Create a Policy Name that is 64 chars (or less) using hash as an option [unique and repeatable]
+                $ShortNameRT = Create-Hash -StringToHash $PolicyResourceDisplayName
 
-                # Make sure export directory exists!
-                if(!(Test-path "$($ExportDir)\$($RPVar[0])"))
+                if($ExportInitiative)
                 {
-                    try
-                    {
-                        $NULL = new-item -ItemType Directory -Path "$($ExportDir)\$($RPVar[0])"
-                    }
-                    catch
+                    $JSONType = @'
+{
+    "type": "Microsoft.Authorization/policyDefinitions",
+    "apiVersion": "2019-09-01",
+    "name": "<SHORT NAME OF SERVICE>",
+
+'@
+                    $PolicyRSID = """[resourceId('Microsoft.Authorization/policyDefinitions/', '$($ShortNameRT)')]"""
+                    $PolicyRSIDs = $PolicyRSIDs + "                "  + $PolicyRSID + "," + "`r`n"
+                    $JSONTYPE = $JSONType.replace("<SHORT NAME OF SERVICE>", "$($ShortNameRT)")
+                    $PolicyJSON = Update-EventHubJSON -resourceType $Type.ResourceType -metricsArray $metricsArray -logsArray $logsArray -nameField $RPVar[1] -JSONType $JSONType -ExportInitiative $ExportInitiative -kind $Type.Kind -PolicyResourceDisplayName $PolicyResourceDisplayName -PolicyName $ShortNameRT
+                    $PolicyBag = $PolicyBag + $PolicyJSON[2] +  ',' + "`r`n"
+                    $PolicyDefParam = @'
+                {
+                    "policyDefinitionId": <PolicyResoureceID>,
+                    <Policy Definition Parameters>
+                },
+'@
+                $PolicyDefParam = $PolicyDefParam.Replace("<PolicyResoureceID>", "$($PolicyRSID)")
+                $PolicyDefParam = $PolicyDefParam.Replace("<Policy Definition Parameters>", "$($PolicyJSON[3])")
```

Comment:
I'm having trouble finding where PolicyJSON[3] is instantiated and assigned.  I believe this is the root cause of my failure in attempting the following command.  If this is intended to be parameters should this be PolicyJSON[0]?
`.\Create-AzDiagPolicy.ps1 -ExportAll -ExportEH -ValidateJSON -ExportDir ".\LogPolicies3" -SubscriptionId "<subscriptionId>" -AllRegions -ExportInitiative`